### ISSUE TITLE
FScore with best  threshold calculation 

### ISSUE THREAD

**Issue by PedroCardoso** [tensorflow/addons#2059]
**Describe the feature and the current behavior/state.**
A F1Score metric that allowed us to define N threshold splits. We would calculate values to each threshold, and onn result we would return the best F Score. 
This could be an update to the existing FBetaScore or a new class. On the existing class, we would have a num_thresholds argument to __init__ . result() would work as now, returning the best, and we could add a new method best_threshold() 

**Relevant information**
- Are you willing to contribute it (yes/no): yes
- Are you willing to maintain it going forward? (yes/no): yes
- Is there a relevant academic paper? (if so, where): no
- Is there already an implementation in another framework? (if so, where): It used to exist in TF1
- Was it part of tf.contrib? (if so, where): There was metrics  on TF1 called *at_thresholds that worked similarly , I believe. 

**Which API type would this fall under (layer, metric, optimizer, etc.)**
metric

**Who will benefit with this feature?**
me ! and anyone wanting best F Score at best threshold !

**Any other info.**

**Comment by marload** [comment]
This feature seems useful to me. Personally, it seems better to create a new class than to update FBetaScore. :smile:

**Comment by PedroCardoso** [comment]
I can take a go, but would be happy to have proposals for Class names

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
